packages/mui-material/src/TextField/TextField.d.ts

```diff
@@ -226,7 +226,13 @@ export interface OutlinedTextFieldProps extends BaseTextFieldProps {
   InputProps?: Partial<OutlinedInputProps>;
 }
 
-export type TextFieldProps = StandardTextFieldProps | FilledTextFieldProps | OutlinedTextFieldProps;
+type Variants = 'outlined' | 'standard' | 'filled';
```

Comment:
One more thing—we should export this to avoid issues like #36035, #33165 for library apps. See their fixes in #33751 and #36057. More details in https://github.com/mui/material-ui/issues/36035#issuecomment-1415895742.
```suggestion
export type Variants = 'outlined' | 'standard' | 'filled';
```

Reply:
got it, i'll export. Should variable name be changed to something like `TextFieldVariants` ?

Reply:
Sounds good.